lib/synapse/service_watcher/zookeeper_aurora.rb

```diff
@@ -0,0 +1,76 @@
+# encoding: utf-8
+#
+# Variant of the Zookeeper service watcher that works with Apache Aurora
+# service announcements.
+#
+# Parameters:
+#   hosts: list of zookeeper hosts to query (List of Strings, required)
+#   path: "/path/to/serverset/in/zookeeper" (String, required)
+#   port_name: Named service endpoint (String, optional)
+#
+# If port_name is omitted, uses the default serviceEndpoint port.
+
+# zk node data looks like this:
+#
+# {
+#   "additionalEndpoints": {
+#     "aurora": {
+#       "host": "somehostname",
+#       "port": 31943
+#     },
+#     "http": {
+#       "host": "somehostname",
+#       "port": 31943
+#     },
+#     "otherport": {
+#       "host": "somehostname",
+#       "port": 31944
+#     }
+#   },
+#   "serviceEndpoint": {
+#     "host": "somehostname",
+#     "port": 31943
+#   },
+#   "shard": 0,
+#   "status": "ALIVE"
```

Comment:
Looking at Aurora, I am not sure if the status is ever toggled. Aurora seems to simply remove the znode when the registered server disconnects. I'd say we leave it out for now.